### PR TITLE
Add `accessDeniedWhen`

### PR DESCRIPTION
**Pull request by amitaibu**
My brain can't work with the double negative of `accessDeniedUnless`. It's much easier for me to say when to  access deny.

### CODE REVIEW

**Comment by amitaibu** [comment]
Off-topic to this PR, but I noticed the status code returned is `500`, but it should be 403

image

**Comment by mpscholten** [comment]
Thanks 👍  Can you add this to the docs as well https://ihp.digitallyinduced.com/Guide/recipes.html#checking-that-the-current-user-has-permission-to-access-the-action ?

> Off-topic to this PR, but I noticed the status code returned is 500, but it should be 403

Good point, can you open an issue re this?

**Comment by amitaibu** [comment]
Added docs.

New issue created - https://github.com/digitallyinduced/ihp/issues/1734

File: IHP/AuthSupport/Authorization.hs [hunk 1]
```diff
@@ -10,16 +10,31 @@ import IHP.Prelude
...
+-- 
+-- > action EditPostAction { postId } = do
+-- >     post <- fetch postId
+-- >     accessDeniedWhen (post.authorId /= currentUserId)
+-- >     
+-- >     renderHtml EditView { .. }
+--
+-- This will throw an error and prevent the view from being rendered when the current user is not the author of the post.
+accessDeniedWhen :: Bool -> IO ()
+accessDeniedWhen condition = if condition then fail "Access denied" else pure ()
```
Comment (amitaibu) [inline IHP/AuthSupport/Authorization.hs:25]:
```suggestion
accessDeniedWhen condition = when condition (fail "Access denied")
```

File: IHP/AuthSupport/Authorization.hs [hunk 2]
```diff
@@ -10,16 +10,31 @@ import IHP.Prelude
...
 -- >     accessDeniedUnless (post.authorId == currentUserId)
 -- >     
 -- >     renderHtml EditView { .. }
 --
--- This will throw an error and prevent the view from being rendered when the current user is not author of the post.
+-- This will throw an error and prevent the view from being rendered when the current user is not the author of the post.
 accessDeniedUnless :: Bool -> IO ()
-accessDeniedUnless condition = if condition then pure () else fail "Access denied"
\ No newline at end of file
+accessDeniedUnless condition = if condition then pure () else fail "Access denied"
```
Comment (amitaibu) [inline IHP/AuthSupport/Authorization.hs:39]:
```suggestion
accessDeniedUnless condition = unless condition (fail "Access denied")
```

**Review by mpscholten — APPROVED** [review]
Thanks 👍

**Pull request merged**